etl/tests/embedded/test_basement.py

```diff
@@ -665,3 +667,252 @@ def test_basement_to_dict(sample_basement_element: element.Element) -> None:
+    """
+    <FloorHeader adjacentEnclosedSpace="data" id="data">
+        <Label>BW hdr-01</Label>
+        <Construction>
+            <Type rValue="data" nominalInsulation="data">0000000000</Type>
+        </Construction>
+        <Measurements height="data" perimeter="data" />
+        <FacingDirection code="data">
+            <English>N/A</English>
+            <French>S/O</French>
+        </FacingDirection>
+    </FloorHeader>
+    """,
+
+    # This XML block is missing the insulation attributes of the Type tag
+    """
+    <FloorHeader adjacentEnclosedSpace="false" id="3">
+        <Label>BW hdr-01</Label>
+        <Construction>
+            <Type >0000000000</Type>
+        </Construction>
+        <Measurements height="0.23" perimeter="34.7456" />
+        <FacingDirection code="1">
+            <English>N/A</English>
+            <French>S/O</French>
+        </FacingDirection>
+    </FloorHeader>
+    """
+]
+
+
+BAD_FOUNDATION_XML = [
+    # This XML block has an unknown foundation type tag
+    """
+    <SomeWeirdFoundation isExposedSurface="true" exposedSurfacePerimeter="34.7466" id="1">
+        <Label>Foundation - 2</Label>
+        <Configuration type="BBEN" subtype="1" overlap="0">BCIN_1</Configuration>
+    </SomeWeirdFoundation>
+    """,
+
+    # This XML block is missing the type attribute of teh Configuration tag
+    """
+    <Basement isExposedSurface="true" exposedSurfacePerimeter="34.7466" id="1">
+        <Label>Foundation - 2</Label>
+        <Configuration subtype="1" overlap="0">BCIN_1</Configuration>
+    </Basement>
+    """,
+
+    # This XML block is missing it's Label tag
+    """
+    <Crawlspace isExposedSurface="true" exposedSurfacePerimeter="9.144" id="29">
+        <Configuration type="SCN" subtype="1">SCN_1</Configuration>
+    </Crawlspace>
+    """,
+
+    # This XML block has an empty string as the type attribute of the Configuration tag
+    """
+    <Crawlspace isExposedSurface="true" exposedSurfacePerimeter="9.144" id="29">
+        <Label>Crawl</Label>
+        <Configuration type="" subtype="1">SCN_1</Configuration>
+    </Crawlspace>
+    """
+]
+
+
+@pytest.mark.parametrize("bad_xml", BAD_FOUNDATION_FLOOR_XML)
+def test_bad_basement_floor(bad_xml: str) -> None:
+    node = element.Element.from_string(bad_xml)
+    with pytest.raises(InvalidEmbeddedDataTypeError) as excinfo:
+        basement.BasementFloor.from_basement(node)
+
+    assert excinfo.value.data_class is basement.BasementFloor
+
+
+@pytest.mark.parametrize("bad_xml", BAD_FOUNDATION_FLOOR_XML)
+def test_bad_crawlspace_floor(bad_xml: str) -> None:
+    node = element.Element.from_string(bad_xml)
+    with pytest.raises(InvalidEmbeddedDataTypeError) as excinfo:
+        basement.BasementFloor.from_crawlspace(node)
+
+    assert excinfo.value.data_class is basement.BasementFloor
+
+
+
+@pytest.mark.parametrize("bad_xml", BAD_FOUNDATION_CRAWLSPACE_WALL_XML)
+def test_bad_crawlspace_wall(bad_xml: str) -> None:
```

Comment:
Better way to do this:
- define a function that return a list of tuples of (Bad XML data, data_class)
- define a single test function that uses this function to parametrize the test
- run the first element of the parametrized tuple through the test, assert that the resulting data_class is the second element